js/date-filter.js

```diff
@@ -125,6 +146,127 @@ DateFilter.prototype.handleModifyEvent = function(e, opts) {
   }
 };
 
+DateFilter.prototype.setupDateGrid = function() {
+  var dateBegin = this.$minDate.val().split('/');
+  var dateEnd = this.$maxDate.val().split('/');
+  var dateRangeFirst = this.$grid.find('.date-range-first');
+  var dateRangeSecond = this.$grid.find('.date-range-second');
+  var minDateMonth = dateBegin[0];
+  var minDateYear = dateBegin[2];
+  var maxDateMonth = dateEnd[0];
+  var maxDateYear = dateEnd[2];
+  var transactionYear = $('#two-year-transaction-period').val();
```

Comment:
This value is currently set as a property by the `handleModifyEvent` method, which is called when the transaction period changes. So I think you should be able to just reference it from the `this.maxYear` property.